README.md

```diff
@@ -97,179 +97,225 @@ import ballerina/log;
-    username: config:getAsString("DATABASE_USERNAME", default = "root"),
-    password: config:getAsString("DATABASE_PASSWORD", default = "root"),
-    dbOptions: { useSSL: false }
-};
-
-endpoint http:Listener listener {
-    port: 9090
-};
+// Create SQL client for MySQL database
+mysql:Client employeeDB = new({
+        host: config:getAsString("DATABASE_HOST", default = "localhost"),
+        port: config:getAsInt("DATABASE_PORT", default = 3306),
+        name: config:getAsString("DATABASE_NAME", default = "EMPLOYEE_RECORDS"),
+        username: config:getAsString("DATABASE_USERNAME", default = "root"),
+        password: config:getAsString("DATABASE_PASSWORD", default = "123"),
+        dbOptions: { useSSL: false }
+    });
 
 // Service for the employee data service
 @http:ServiceConfig {
     basePath: "/records"
 }
-service<http:Service> EmployeeData bind listener {
+service EmployeeData on httpListener {
 
     @http:ResourceConfig {
         methods: ["POST"],
         path: "/employee/"
     }
-    addEmployeeResource(endpoint httpConnection, http:Request request) {
+    resource function addEmployeeResource(http:Caller httpCaller, http:Request request) {
         // Initialize an empty http response message
-        http:Response response;
-        Employee employeeData;
+        http:Response response = new;
+
         // Extract the data from the request payload
-        var payloadJson = check request.getJsonPayload();
-        employeeData = check <Employee>payloadJson;
-
-        // Check for errors with JSON payload using
-        if (employeeData.name == "" || employeeData.age == 0 || employeeData.ssn == 0 ||
-            employeeData.employeeId == 0) {
-            response.setTextPayload("Error : json payload should contain
-             {name:<string>, age:<int>, ssn:<123456>,employeeId:<int>} ");
-            response.statusCode = 400;
-            _ = httpConnection->respond(response);
-            done;
+        var payloadJson = request.getJsonPayload();
+
+        if (payloadJson is json) {
+            Employee|error employeeData = Employee.convert(payloadJson);
+
+            if (employeeData is Employee) {
+                // Validate JSON payload
+                if (employeeData.name == "" || employeeData.age == 0 || employeeData.ssn == 0 ||
+                    employeeData.employeeId == 0) {
+                    response.setPayload("Error : json payload should contain
+                    {name:<string>, age:<int>, ssn:<123456>, employeeId:<int>}");
+                    response.statusCode = 400;
+                } else {
+                    // Invoke insertData function to save data in the MySQL database
+                    json ret = insertData(employeeData.name, employeeData.age, employeeData.ssn,
+                        employeeData.employeeId);
+                    // Send the response back to the client with the employee data
+                    response.setJsonPayload(ret);
```

Comment:
Shall we use setPayload here  since we encourage users to use the generic method instead of specific ones.